sklearn/mixture/gaussian_mixture.py

```diff
@@ -0,0 +1,677 @@
+
+    init_params : {'kmeans', 'random'}, defaults to 'kmeans'.
+        The method used to initialize the weights, the means and the
+        covariances.
+        Must be one of::
+        'kmeans' : responsibilities are initialized using kmeans.
+        'random' : responsibilities are initialized randomly.
+
+    weights_init : array-like, shape (n_components, ), optional
+        The user-provided initial weights, defaults to None.
+        If it None, weights are initialized using the 'init_params' method.
+
+    means_init: array-like, shape (n_components, n_features), optional
+        The user-provided initial means, defaults to None,
+        If it None, means are initialized using the 'init_params' method.
+
+    covariances_init: array-like, optional.
+        The user-provided initial covariances, defaults to None.
+        If it None, covariances are initialized using the 'init_params' method.
+        The shape depends on 'covariance_type'::
+            (n_components,)                        if 'spherical',
+            (n_features, n_features)               if 'tied',
+            (n_components, n_features)             if 'diag',
+            (n_components, n_features, n_features) if 'full'
+
+    random_state: RandomState or an int seed, defaults to None.
+        A random number generator instance.
+
+    warm_start : bool, default to False.
+        If 'warm_start' is True, the solution of the last fitting is used as
+        initialization for the next call of fit(). This can speed up
+        convergence when fit is called several time on similar problems.
+
+    verbose : int, default to 0.
+        Enable verbose output. If 1 then it prints the current
+        initialization and each iteration step. If greater than 1 then
+        it prints also the log probability and the time needed
+        for each step.
+
+    Attributes
+    ----------
+    weights_ : array, shape (`n_components`,)
+        The weights of each mixture components.
+        `weights_` will not exist before a call to fit.
+
+    means_ : array, shape (`n_components`, `n_features`)
+        The mean of each mixture component.
+        `means_` will not exist before a call to fit.
+
+    covariances_ : array
+        The covariance of each mixture component.
+        The shape depends on `covariance_type`::
+            (n_components,)                        if 'spherical',
+            (n_features, n_features)               if 'tied',
+            (n_components, n_features)             if 'diag',
+            (n_components, n_features, n_features) if 'full'
+        `covariances_` will not exist before a call to fit.
+
+    converged_ : bool
+        True when convergence was reached in fit(), False otherwise.
+        `converged_` will not exist before a call to fit.
+
+    initialized_ : bool
+        True when fit() has been applied at least once, False otherwise.
+        `initialized_` will not exist before a call to fit.
```

Comment:
in the end is it necessary to have initialized_ param? as for example covariances_ will not exist if fit was not called. So you already know if it was initialized_ with other attributes no?